StyleCop.Analyzers/StyleCop.Analyzers/Helpers/MemberOrderHelper.cs

```diff
@@ -113,63 +127,42 @@ private enum ModifierFlags
         internal MemberDeclarationSyntax Member { get; }
 
         /// <summary>
-        /// The priority for this member.
-        /// </summary>
-        /// <value>
-        /// The priority for this member.
-        /// </value>
-        internal int Priority
-        {
-            get
-            {
-                var priority = this.ModifierPriority;
-
-                // accessibility is more important than the modifiers
-                priority += (this.AccessibilityPriority + 1) * 100;
-
-                // element type is more important than accessibility
-                priority += (this.elementPriority + 1) * 1000;
-                return priority;
-            }
-        }
-
-        /// <summary>
-        /// The priority for this member only from accessibility.
-        /// </summary>
-        /// <value>
-        /// The priority for this member.
-        /// </value>
-        internal int AccessibilityPriority => (int)this.accessibilty;
-
-        /// <summary>
-        /// The priority for this member only from modifiers.
+        /// Gets the priority for this member.
         /// </summary>
         /// <value>
         /// The priority for this member.
         /// </value>
-        internal int ModifierPriority => (int)this.modifierFlags;
+        internal int Priority { get; }
 
-        private static ModifierFlags GetModifierFlags(SyntaxTokenList syntax, ElementOrderingChecks checks)
+        internal static AccessLevel GetAccessLevelForOrdering(SyntaxNode member, SyntaxTokenList modifiers)
         {
-            var flags = ModifierFlags.None;
-            if (checks.Const && syntax.Any(SyntaxKind.ConstKeyword))
+            SyntaxKind type = member.Kind();
+
+            AccessLevel accessibility;
+            if ((type == SyntaxKind.ConstructorDeclaration && modifiers.Any(SyntaxKind.StaticKeyword))
+                || (type == SyntaxKind.MethodDeclaration && ((MethodDeclarationSyntax)member).ExplicitInterfaceSpecifier != null)
+                || (type == SyntaxKind.PropertyDeclaration && ((PropertyDeclarationSyntax)member).ExplicitInterfaceSpecifier != null)
+                || (type == SyntaxKind.IndexerDeclaration && ((IndexerDeclarationSyntax)member).ExplicitInterfaceSpecifier != null))
```

Comment:
:arrow_right: Can you file this as a separate bug? For this PR I just reused the code from #1180/#1179. So far it hasn't been reported as a deviation from StyleCop Classic so I'd like to get a bit more discussion to ensure the change is intentional and properly documented.